dom-event-delegation/main.ts

```diff
@@ -0,0 +1,17 @@
+const $taskList = document.querySelector('.task-list');
+if (!$taskList) throw new Error('task-list query failed');
+
+function callBack(event: Event): void {
+  const eventTarget = event.target as HTMLElement;
+  console.log('eventTarget:', eventTarget);
+  console.log('eventTarget.tagName:', eventTarget.tagName);
+  if (eventTarget.tagName === 'BUTTON') {
+    const closestTaskListItem = eventTarget.closest(
+      '.task-list-item'
+    ) as HTMLElement;
+    console.log('eventTarget closest task list item:', closestTaskListItem);
+    closestTaskListItem.remove();
+  }
+}
```

Comment:
As a recommendation, I use something called "guard clauses." It's a way of exiting a function if my condition is _not_ met. It's save a minuscule amount of time and but also makes the code more readable.

```
if ($eventTarget.tagName !== 'BUTTON') {
    return;
  }

  const $taskListItem = $eventTarget.closest('.task-list-item');

  console.log('closest .task-list-item', $taskListItem);
  $taskListItem?.remove();
```